### PR TITLE
Tag decoding and spliting

### PR DESCRIPTION
**Pull request by hez**

### CODE REVIEW

**Comment by stewart** [comment]
This feels fairly tightly coupled to a specific way of working with tags, perhaps we can support both "standard" tags as well as the "key::value" format with similar APIs?

i.e.:

```elixir
alias ShopifyAPI.Rest.Tag

Tag.decode("Emotive, Flash Memory, MP3, Music")
# => ["Emotive", "Flash Memory", "MP3", "Music"]

Tag.decode("brand::Kingston, category::MP3 Player, batteries::AA", "::")
# => [{"brand", "Kingston"}, {"category", "MP3 Player"}, {"batteries", "AA"}]

Tag.encode(["Emotive", "Flash Memory", "MP3", "Music"])
# => "Emotive,Flash Memory,MP3,Music"

Tag.encode([{"brand", "Kingston"}, {"category", "MP3 Player"}, {"batteries", "AA"}], "::")
# => "brand::Kingston,category::MP3 Player,batteries::AA"
```

**Comment by hez** [comment]
@stewart decode! and encode! already do that, the split is there to support "complicated" tags, it is configurable as to what string splitting you want to use

File: test/shopify_api/rest/tag_test.exs [hunk 1]
```diff
@@ -0,0 +1,32 @@
+defmodule ShopifyAPI.REST.TagTest do
+  use ExUnit.Case
+  use ExUnitProperties
+
+  alias ShopifyAPI.REST.Tag
+
+  describe "Tags.encode and split" do
```
Comment (stewart) [inline test/shopify_api/rest/tag_test.exs]: This `describe` block needs to be renamed, and can probably be folded in with the other one.

Comment (hez) [reply test/shopify_api/rest/tag_test.exs]: oh, yea! good catch!

**Pull request merged**